File: src/main/java/com/alvinalexander/cato/Field.java
package com.alvinalexander.cato;

/**
 * This class is deprecated because I'd like to move it to Scala, but I haven't been
 * able to get the Scala version to play well with FreeMarker (or vice-versa).
 */
@Deprecated
public class Field {

    String fieldName;
    String fieldNameAsLabel;
    String camelCaseFieldName;
    String javaFieldType;
    String jsonFieldType;
    String phpFieldType;
    String playFieldType;
    String playOptionalFieldType;
    String scalaFieldType;
    String senchaFieldType;
    String databaseFieldType;
    boolean isRequired;

    public Field (
        String fieldName, 
        String fieldNameAsLabel,
        String camelCaseFieldName, 
        String javaFieldType,
        String jsonFieldType,
        String phpFieldType,
        String playFieldType,
        String playOptionalFieldType,
        String scalaFieldType,
        String senchaFieldType,
        String databaseFieldType, 
        boolean isRequired
    ) {
        this.fieldName = fieldName;
        this.fieldNameAsLabel = fieldNameAsLabel;
        this.camelCaseFieldName = camelCaseFieldName;
        this.javaFieldType = javaFieldType;
        this.jsonFieldType = jsonFieldType;
        this.phpFieldType = phpFieldType;
        this.playFieldType = playFieldType;
        this.playOptionalFieldType = playOptionalFieldType; 
        this.scalaFieldType = scalaFieldType;
        this.senchaFieldType = senchaFieldType;
        this.databaseFieldType = databaseFieldType;
        this.isRequired = isRequired;
    }
    
    public String getFieldName() {
        return fieldName;
    }

    public String getFieldNameAsLabel() {
        return fieldNameAsLabel;
    }

    public boolean isRequired() {
        return isRequired;
    }

    public String getCamelCaseFieldName() {
        return camelCaseFieldName;
    }

    public String getDatabaseFieldType() {
        return databaseFieldType;
    }

    public String getJavaFieldType() {
        return javaFieldType;
    }

    public String getJsonFieldType() {
        return jsonFieldType;
    }

    public String getPhpFieldType() {
        return phpFieldType;
    }

    public String getPlayFieldType() {
        return playFieldType;
    }

    public String getPlayOptionalFieldType() {
        return playOptionalFieldType;
    }

    public String getScalaFieldType() {
        return scalaFieldType;
    }

    public String getSenchaFieldType() {
      return senchaFieldType;
    }

}

File: src/main/scala/com/alvinalexander/cato/model/Field.scala
package com.alvinalexander.cato.model

import scala.reflect.BeanProperty

/**
 * TODO i tried to get this class to work with FreeMarker using the usual
 * JavaBean recipe, but I still got errors in FreeMarker, with it complaining
 * abou the cameCaseFieldName.
 * So, I'm currently using the com.alvinalexander.cato.Field class instead of
 * this class.
 */
//class Field(
//    @BeanProperty var fieldName: String,
//    @BeanProperty var camelCaseFieldName: String,
//    @BeanProperty var fieldType: String,
//    @BeanProperty var javaFieldType: String,
//    @BeanProperty var jsonFieldType: String,
//    @BeanProperty var phpFieldType: String,
//    @BeanProperty var playFieldType: String,
//    @BeanProperty var playOptionalFieldType: String,
//    @BeanProperty var scalaFieldType: String,
//    @BeanProperty var databaseFieldType: String,
//    @BeanProperty var isRequired: Boolean
//)
//
//object FieldDriver extends App {
//  
//    val f = new Field(
//            "fieldName",
//            "camel",
//            "fieldType",
//            "javaField",
//            "json",
//            "php",
//            "play",
//            "playOpt",
//            "scala",
//            "db",
//            true
//            )
//    
//    println(f.getCamelCaseFieldName())
//  
//}
